vscode/test/fixtures/mock-server.ts

```diff
@@ -87,10 +89,31 @@ export async function run<T>(around: () => Promise<T>): Promise<T> {
         res.send(`event: completion\ndata: {"completion": ${JSON.stringify(response)}}\n\nevent: done\ndata: {}\n\n`)
     })
 
-    let isFirstCompletion = true
     app.post('/.api/completions/code', (req, res) => {
-        const response = isFirstCompletion ? responses.firstCode : responses.code
-        isFirstCompletion = false
+        const OPENING_CODE_TAG = '<CODE5711>'
+        const request = req as MockRequest
+
+        // Extract the code from the last message.
+        let completionPrefix = request.body.messages.at(-1)?.text
+        if (completionPrefix?.startsWith(OPENING_CODE_TAG)) {
+            completionPrefix = completionPrefix.slice(OPENING_CODE_TAG.length)
+        }
+        // Trim to the last word since our mock responses are just triggered by a word.
+        completionPrefix = completionPrefix?.split(/\s/g).at(-1)
```

Comment:
Splitting on `\s` isn't a great way to look for words because -1 can be a space if you have a trailing space.

```suggestion
        completionPrefix = completionPrefix?.split(/\W+/g).filter(x => x).at(-1)
```

Reply:
I think we care about trailing spaces here though. If my prefix is `"abcd "` I wouldn't expect the server to return results like `"abcdef"` because I've explicitly typed that space (and if I understand correctly, we can't delete it when you accept the completion). The mock implementation was intended to just complete partial words for simplicity.

Reply:
Makes perfect sense.

Consider tweaking the comment along those lines? "or empty string if there's trailing space", something like that, more succinctly if you can manage it?

Reply:
I'm not sure if it classes as succinct, but I've updated the comment to be clearer :)